### PR TITLE
F/scan qr code

### PR DESCRIPTION
**Pull request by Benjamin-Davies**

### CODE REVIEW

**Comment by Benjamin-Davies** [comment]
QR code

Here is a QR code to test the camera with. This one links to https://example.com/challenge/1, but one linking to https://explorer-trial-ui.herokuapp.com/challenge/1 would also work. The QR codes match the url for the challenge so that if a user scans one using their own QR code scanner then the app will still work.

File: stem-explorer-ng/src/app/containers/camera/camera.component.ts [hunk 1]
```diff
@@ -1,15 +1,64 @@
...
+        challenge,
+        name: location?.name,
+        link: location?.link,
+      },
+      panelClass: 'app-dialog',
+    });
+  }
+
+  ngOnInit() {
+    setTimeout(() => {
```
Comment (kaciakmaciak) [inline stem-explorer-ng/src/app/containers/camera/camera.component.ts:50]: There is an undocumented(!) output you can use called `permissionResponse` which has `$event` of type `boolean` (if `true` the app has access to camera, `false` no access to camera). Have a look in their demo if you want more info https://github.com/zxing-js/ngx-scanner/tree/master/projects/zxing-scanner-demo/src/app

File: stem-explorer-ng/src/app/containers/camera/camera.component.ts [hunk 2]
```diff
@@ -1,15 +1,80 @@
...
+    }
+
+    const challengeId = parseInt(match[1], 10);
+    const challenges = await this.api.getChallenges().toPromise();
+    const challenge = challenges.challenges.find((c) => c.uid === challengeId);
+    const locations = await this.api.getLocations().toPromise();
+    const location = locations.location.find((l) => l.uid === challengeId);
+
+    this.dialog.closeAll();
+    this.dialog.open(ListViewDialogComponent, {
```
Comment (kaciakmaciak) [inline stem-explorer-ng/src/app/containers/camera/camera.component.ts:51]: FYI: I think once you merge develop you'll see this component is missing. It's been renamed to `ChallengeDialogComponent` (sorry) and all you need to send in `data` is challenge id.

**Comment by kaciakmaciak** [comment]
@Benjamin-Davies please merge develop before merging.

**Comment by kellymarjorie** [comment]
@Benjamin-Davies looks like you have merged develop in; can you mark the change request as being completed so we can merge this branch in? Thanks

**Comment by Benjamin-Davies** [comment]
@kellymarjorie I don't think I can. I think @kaciakmaciak neeeds to as she's the one who requested them.

**Pull request merged**